lib/puppet/property.rb

```diff
@@ -587,7 +588,10 @@ def validate_features_per_value(value)
     if features = self.class.value_option(self.class.value_name(value), :required_features)
       features = Array(features)
       needed_features = features.collect { |f| f.to_s }.join(", ")
-      raise ArgumentError, "Provider #{provider.class.name} must have features '#{needed_features}' to set '#{self.class.name}' to '#{value}'" unless provider.satisfies?(features)
+      unless provider.satisfies?(features)
+        raise ArgumentError, _("Provider %{provider} must have features '%{needed_features}' to set '%{property}' to '%{value}'") %
```

Comment:
`Provider` is a class, but it's also terminology. Not sure whether it should be translated or not in this case?

Reply:
Here is what I could find on confluence https://confluence.puppetlabs.com/display/enterprise/Translation-friendly+strings#Translation-friendlystrings-Commandsandtechnicalterminology